### DISCUSSION TITLE
Request for AES128_CBC Encryption Help

### DISCUSSION THREAD

**Discussion by tecsantoshkumar** [openssl/openssl#23909] [Q&A]
I am currently working on implementing AES128_CBC encryption on an STM32 embedded system. I have encountered some difficulties with incorporating salt and iteration counts into the encryption process.

I would greatly appreciate it if you could provide some guidance or suggestions on how to correctly implement AES128_CBC encryption with salt and iteration counts with or without OpenSSL library on an STM32 embedded system.

Thank you very much for your time and assistance. I look forward to hearing from you soon.

**Comment by tom-cosgrove-arm** [comment]
This is not a bug in OpenSSL, and this issue should be converted to a discussion (I don't have rights to do that).

In that discussion, it will probably be mentioned that AES-CBC is a mode of symmetric encryption; these generally do not have salt or iteration counts. Salt and iteration count together are most commonly found in PBKDF2, which uses hash functions, not symmetric encryption.

Perhaps in a discussion you could elaborate on your requirements beyond "incorporate salt and iteration count with AES128-CBC" - for example, why you think this would be necessary
